File: API/Models/Message.cs
using System;

namespace API.Models;

public class Message
{
    public int Id { get; set; }
    public string? SenderId { get; set; } = string.Empty;
    public string? ReceiverId { get; set; } = string.Empty;
    public string? Content { get; set; } = string.Empty;
    public DateTime CreatedDate { get; set; }
    public bool IsRead { get; set; }
    public AppUser? Receiver { get; set; }
    public AppUser? Sender { get; set; }
}

File: client/src/app/models/message.ts
export interface Message {
    id: number;
    senderId: string | null;
    receiverId: string | null;
    content: string | null;
    attachmentUrl?: string | null;
    attachmentType?: 'image' | 'file' | null;
    attachmentName?: string | null;
    createdDate: string;
    isRead: boolean;
}